File: publisher.rb
require "rubygems"
require "bunny"
require "json"

conn = Bunny.new "amqp://tvobttnq:Ri6S8XroB9gRiIvSesqQMa2KcWI_IOkB@otter.rmq.cloudamqp.com/tvobttnq"
conn.start


ch = conn.create_channel

q=ch.queue("examplequeue", :durable=> true)
x = ch.direct("example.exchange", :durable => true)

q.bind(x, :routing_key => "process")

mesg = {:name => "tom", :mail => "asdsa@asdasd.com", :size => 10}

(1..20).each do |item|
    x.publish(JSON.generate(mesg), :timestamp=> Time.now.to_i, :routing_key=>"process")
    sleep 1
end


sleep 3.0
conn.close

File: publisher.js
var _ = require("lodash");

const q = "tasks";

async function amqp() {
  try {
    console.log("[*] Connecting");
    let con = await require("amqplib").connect(
      "amqp://tvobttnq:Ri6S8XroB9gRiIvSesqQMa2KcWI_IOkB@otter.rmq.cloudamqp.com/tvobttnq"
    );

    let ch = await con.createChannel();
    let _ok = await ch.assertQueue(q, { durable: false });
    console.log(`[*] sending Hello message`);
    ch.sendToQueue(q, Buffer.from("Hello!"));

    _.range(10).forEach(m => {
      console.log(`[*] sending message ${m} from loop`);
      ch.sendToQueue(q, Buffer.from("[*] message n: " + m));
    });

    console.log("[*] messages were sended");
    return con;
    //con.close();
  } catch (error) {
    console.log("ERR:" + error);
  }
}

/* (async function() {
  await amqp();
})(); */

amqp().then(con => {
  setTimeout(function() {
    con.close();
    process.exit(0);
  }, 500);
});
